python/ray/dag/compiled_dag_node.py

```diff
@@ -563,7 +563,7 @@ def __init__(
             )
 
         self._enable_asyncio: bool = enable_asyncio
-        self._fut_queue = asyncio.Queue()
+        self._fut_queues: Dict[ChannelInterface, asyncio.queues.queue] = {}
```

Comment:
Note: `self._result_buffer` and perhaps `self._dag_output_fetchers` need refactoring.

Reply:
`self._fut_queues` and `self._dag_output_fetchers` are changed to lists with indexing consistent with `self.dag_output_channels`. `self._result_buffer` is still a dict but no longer uses `ChannelInterface` for indexing (instead, `{execution_index->{channel_index->result}}`). The reason I kept it as a dict is that when we get and remove results from the buffer, the remaining results won't be out of order.

An alternative to use list for `self._result_buffer` is to mark the popped results as `None` instead of remove them from the buffer. However, the downside is that the size of the list remains the same even if we get all results from the buffer. From debugging perspective, I don't see a clear advantage for one over the other. Let me know if you prefer `self._result_buffer` to be a list as well.